wcfsetup/install/files/lib/system/form/builder/field/UploadFormField.class.php

```diff
@@ -340,52 +340,117 @@ public function getFieldHtml()
     /**
      * @inheritDoc
      *
-     * @throws \InvalidArgumentException    if the getter for the value provides invalid values
+     * @throws \InvalidArgumentException    if the getter for the value or the object's property provides invalid values
      */
     public function updatedObject(array $data, IStorableObject $object, $loadValues = true)
     {
         if ($loadValues) {
-            // first check, whether an getter for the field exists
-            if (\method_exists($object, 'get' . \ucfirst($this->getObjectProperty()) . 'UploadFileLocations')) {
-                $value = \call_user_func([
-                    $object,
-                    'get' . \ucfirst($this->getObjectProperty()) . 'UploadFileLocations',
-                ]);
-                $method = "method '" . \get_class($object) . "::get" . \ucfirst($this->getObjectProperty()) . "UploadFileLocations()'";
-            } elseif (\method_exists($object, 'get' . \ucfirst($this->getObjectProperty()))) {
-                $value = \call_user_func([$object, 'get' . \ucfirst($this->getObjectProperty())]);
-                $method = "method '" . \get_class($object) . "::get" . \ucfirst($this->getObjectProperty()) . "()'";
+            $propertyName = \ucfirst($this->getObjectProperty());
+            $className = static::class;
```

Comment:
Previously, the Variable `className` contains the classname of the given object, not the class of the `UploadFormField`.